### ISSUE TITLE
[BUG] Accessibility issue reported even though parent has `aria-hidden=true`

### ISSUE THREAD

**Issue by nikolay-g** [IBMa/equal-access#208]
## What tool(s) are you using?

- [ x] `accessibility-checker-engine`
- [x ] `accessibility-checker`

## Detailed description

I have been scanning a Material-UI app for accessibility issues with the [IBM Accessibility Chrome plugin](https://www.ibm.com/able/toolkit/tools/) in WCAG 2.1 (A,AA) compliance mode. The tool found an issue which can even be reproduced/detected on the [official Material-UI demo page](https://material-ui.com/components/text-fields/#outlined-input-adornments).

The issue says that `<fieldset> element does not have a <legend>`. The corresponding HTML is:

```html
<fieldset aria-hidden="true" class="jss193 MuiOutlinedInput-notchedOutline">
    <!--- Has style "visibility: hidden" --->
    <legend class="jss195">
        <span>Outlined</span>
    </legend>
</fieldset>
```
The `legend` has style `visibility: hidden` coming from the CSS class. However, the parent `fieldset` has `aria-hidden="true"` attribute and I therefore it should be ignore by readers and assistive technology. I also tried adding the `role="presentation"` attribute to the `fiedlset`, but the issue was still reported by the tool.

I've also raised the issue with the Material-UI team [here](https://github.com/mui-org/material-ui/issues/22387).

Below are screenshots of the problem:
![visibility-hidden](https://user-images.githubusercontent.com/25423557/91671442-1e8f7400-eb6a-11ea-82d0-ecc6f145e37f.png)
![accessibility-issue](https://user-images.githubusercontent.com/25423557/91671444-22bb9180-eb6a-11ea-9d0f-9cf98aef453d.png)

## Steps to reproduce the issue

1. Step one - Open the [official Material-UI demo page](https://material-ui.com/components/text-fields/#outlined-input-adornments).
2. Step two - run a scan with the IBM Chrome Accessibility tool in  WCAG 2.1 (A,AA) compliance mode.

**Comment by tombrunet** [comment]
Thank you for this report. We are looking through various issues related to aria-hidden. They have to be looked at somewhat on a case by case basis since some items marked as aria-hidden can impact users beyond just screen reader users - for instance, color contrast issues on an aria-hidden element are still a problem for users with color blindness - so we cannot just blanket ignore issues within aria-hidden elements. Before we determine that it's safe to disable, we want to make sure that we've thought through the various scenarios to ensure that we're not allowing impacts to end users.

**Comment by MHoov** [comment]
As of June 1
- [x] general rule description (SME & Dev)
- [ ] testcases written (Dev)
- [ ] rule developed  (Dev)
- [ ] message written (SME)
- [ ] help written (SME)

**Comment by marcjohlic** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Equal-Access-61ae71f9f8b4ec001e136462/issues/IBMa/equal-access/208?planning-poker) with ZenHub @aliunwala @mbgower @shunguoy @tombrunet @philljenkins @drjoho

**Comment by shunguoy** [comment]
requested review by @tombrunet
